lib/downloader.js

```diff
@@ -23,65 +26,79 @@ function download(link, callback) {
       status = new Error('Response Error: ' + res.statusCode);
       return;
     }
-    downloadFile.pipe(fs.createWriteStream(filePath, {mode: 493}));
+    downloadFile.pipe(fs.createWriteStream(to, {
+      mode: 493
+    }));
   });
 
-  downloadFile.on('error', function error(err) { callback(err); });
-
-  downloadFile.on('end', function end() { callback(status); });
+  downloadFile.on('error', function error(err) {
+    callback(err);
+  });
 
+  downloadFile.on('end', function end() {
+    callback(status);
+  });
 }
 
 function exec(path) {
   'use strict';
   return (isWin) ? path + '.exe' : path;
 }
 
-function createBase(binDir) {
+function downloader(userBinPath, callback) {
   'use strict';
-  dir = (binDir) ? binDir : defaultBin;
-  mkdirp.sync(dir);
-  if (binDir) { mkdirp.sync(defaultBin); }
-  filePath = path.join(dir, exec('youtube-dl'));
-}
 
-function downloader(binDir, callback) {
-
-  'use strict';
-  if (typeof binDir === 'function') {
-    callback = binDir;
-    binDir = null;
+  if (typeof userBinPath === 'function') {
+    callback = userBinPath;
+    userBinPath = null;
   }
 
-  createBase(binDir);
+  // prepare needed paths
+  binPath = (userBinPath) ? userBinPath : defaultBinPath;
+  filePath = path.join(binPath, exec('youtube-dl'));
+  detailsPath = path.join(binPath, 'details');
+
```

Comment:
TBH, the logic in `createBase` is wrong and the naming of variables are very confusing, so I rewrote that a little bit.